src/Faker/ORM/Doctrine/ColumnTypeGuesser.php

```diff
@@ -36,20 +34,20 @@ public function guessFormat($fieldName, ClassMetadata $class)
                     return $generator->randomNumber($size + 2) / 100;
                 };
             case 'smallint':
-                return function () {
-                    return mt_rand(0, 65535);
+                return function () use ($generator) {
+                    return $generator->numberBetween(0, 65535);
                 };
             case 'integer':
-                return function () {
-                    return mt_rand(0, intval('2147483647'));
+                return function () use ($generator) {
+                    return $generator->numberBetween(0, 2147483647);
                 };
             case 'bigint':
-                return function () {
-                    return mt_rand(0, intval('18446744073709551615'));
+                return function () use ($generator) {
+                    return $generator->numberBetween(0, (int) '18446744073709551615');
```

Comment:
Same here, why not use the raw integer type?

Reply:
Actually I have no idea. I tried that on https://3v4l.org/rSrLB and it seems to produce same results for all PHP versions. Skipping the casting then.